src/main/java/org/apache/commons/io/input/MemoryMappedFileInputStream.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.io.input;
+
+import static org.apache.commons.io.IOUtils.EOF;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileChannel.MapMode;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+import org.apache.commons.io.ByteBufferCleaner;
+
+/**
+ * An {@link InputStream} that utilizes memory mapped files to improve
+ * performance. A sliding window of the file is mapped to memory to avoid
+ * mapping the entire file to memory at one time. The size of the sliding buffer
+ * is user configurable.
+ * 
+ * For most operating systems, mapping a file into memory is more expensive than
+ * reading or writing a few tens of kilobytes of data. From the standpoint of
+ * performance it is generally only worth mapping relatively large files into
+ * memory. Use of this class can provide approximately a 25% increase in
+ * throughput when reading large files. <br>
+ * Note: Use of this class does not necessarily obviate the need to use a
+ * {@link BufferedInputStream}. Depending on the use case, the use of buffering
+ * may still further improve performance. For example:
+ * 
+ * <pre>
+ * new BufferedInputStream(new GzipInputStream(new MemoryMappedFileInputStream(path))))
+ * </pre>
+ * 
+ * will greatly outperform:
+ * 
+ * <pre>
+ * new GzipInputStream(new MemoryMappedFileInputStream(path))
```

Comment:
This comparison does not make sense. You should compare using this class with a code fragment that does not use it.

Reply:
This comment has nothing to do with the performance of this class. It is just noting that user's shouldn't think there is no need for buffering after using this class. In the example given, buffering is needed to improve the performance of the gzip compression in memory.